### PR TITLE
[10.0][FIX] Avoid duplicate validations on picking do_new_transfer

### PR DESCRIPTION
**Pull request by rousseldenis**
Description of the issue/feature this PR addresses:

Stock pickings should not be validated anymore when state is done or cancelled.
If users are filling in stock operations on the same picking at the same time, when user 1 validate picking and then user 2 validate same picking, there are unwanted duplicate moves.

Tested on Runbot 10.0

Current behavior before PR:

Create a picking
Fill in one product with quantity 1
Reserve picking
Fill in quantity done to 1
Save
Open new tab in browser with the same picking
Validate picking
Validate picking on the first tab
Two moves are validated

Desired behavior after PR is merged:

Create a picking
Fill in one product with quantity 1
Reserve picking
Fill in quantity done to 1
Save
Open new tab in browser with the same picking
Validate picking
Validate picking on the first tab
One line of move is validated

OPW-747235

This is quite similar as #17236

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by rousseldenis** [comment]
States are filtered the same way as in the view

**Comment by rousseldenis** [comment]
@jco-odoo Can you look at this and confirm the bug ?

File: addons/stock/models/stock_picking.py [hunk 1]
```diff
@@ -852,11 +852,12 @@ def do_transfer(self):
         """ If no pack operation, we do simple action_done of the picking.
         Otherwise, do the pack operations. """
         # TDE CLEAN ME: reclean me, please
-        self._create_lots_for_picking()
+        pickings = self.filtered(lambda p: p.state not in ['done', 'cancel'])
```
Comment (moylop260) [inline addons/stock/models/stock_picking.py]: What about use just one filtered result?
```python
pickings = self.filtered(lambda p: p.state not in ['done', 'cancel'])
for pick in pickings:
  ...
other_pickings = pickings - no_pack_op_pickings
```

Comment (rousseldenis) [inline addons/stock/models/stock_picking.py]: Yes, that can be more efficient !

**Comment by rousseldenis** [comment]
Fixed by **https://github.com/odoo/odoo/commit/6a732fd2c9de26ee7c4ec03990419cb4c768245**
and https://github.com/odoo/odoo/pull/17799

**Pull request closed without merging**